CleverPush/CleverPush/Source/CleverPush.m

```diff
@@ -1958,6 +1958,60 @@ + (NSArray*)getNotifications {
     return notifications;
 }
 
+#pragma mark - Retrieving notifications based on the flag remote/local
++ (NSArray*)getNotifications:(BOOL)combineWithApi {
+    if (combineWithApi) {
+        NSString *combinedURL = [self createDynamicURLForNotification];
+        NSMutableArray* combinedArray = [[self getNotifications] mutableCopy];
+        NSMutableArray *remoteNotifications = [self getRemoteNotifications:combinedURL];
+        return [combinedArray arrayByAddingObjectsFromArray:remoteNotifications];
+    } else {
+        return [self getNotifications];
+    }
+}
+
+#pragma mark - Creating URL based on the topic dialogue and append the topicId's as a query parameter.
++ (NSString *)createDynamicURLForNotification {
+    NSString *baseURL = [NSString stringWithFormat:@"https://api.cleverpush.com/channel/%@/received-notifications?", channelId];
+    if ([self hasSubscriptionTopics]) {
+        NSMutableArray* dynamicQueryParameters = [self urlQueryParameter];
+        NSString* appendableQueryParameters = [dynamicQueryParameters componentsJoinedByString:@""];
+        NSString *concatenatedURL = [NSString stringWithFormat:@"%@%@", baseURL, appendableQueryParameters];
+        return concatenatedURL;
+    } else {
+        return baseURL;
+    }
+}
+
+#pragma mark - Appending the topicId's as a query parameter.
++ (NSMutableArray*)urlQueryParameter {
```

Comment:
Please rename this method to something descriptive e.g. `getReceivedNotificationsQueryParameters`